examples/consumer.module.ts

```diff
@@ -1,41 +1,30 @@
-import { Module } from "@nestjs/common";
-import { MemphisModule, MemphisService } from "memphis-dev/nest"
-
+import { Inject, Module } from '@nestjs/common';
+import { MemphisModule, MemphisService, consumeMessage } from 'memphis-dev/nest';
+import type { MemphisType } from 'memphis-dev/types';
+import memphis from 'memphis-dev';
 @Module({
-    imports: [MemphisModule.register()],
+    imports: [MemphisModule.register()]
 })
-
 export class ConsumerModule {
-    constructor(private memphis: MemphisService) {
-    }
-
-    startConsumer() {
-        (async function () {
-            let memphisConnection
-            try {
-                memphisConnection = await this.memphis.connect({
-                    host: "<memphis-host>",
-                    username: "<application type username>",
-                    connectionToken: "<broker-token>",
-                });
-
-                const consumer = await memphisConnection.consumer({
-                    stationName: "<station-name>",
-                    consumerName: "<consumer-name>",
-                    consumerGroup: "",
-                });
-
-                consumer.on("message", (message) => {
-                    console.log(message.getData().toString());
-                    message.ack();
-                });
+    // constructor(private memphis: MemphisService) {}
 
-                consumer.on("error", (error) => { });
-            } catch (ex) {
-                console.log(ex);
-                if (memphisConnection)
-                    memphisConnection.close();
-            }
-        })();
+    @consumeMessage({
+        stationName: '<station-name>',
+        consumerName: '<consumer-name>',
+        consumerGroup: ''
+    })
+    async startConsumer(): Promise<MemphisType> {
```

Comment:
this function should get message where is it? + why each time a new message arrived you are connecting to memphis. Connect should be called once